### ISSUE TITLE
ignore i18next t() by default

### ISSUE THREAD

**Issue by AdrienLemaire** [edvardchen/eslint-plugin-i18next#23]
By default, i18next/no-literal-string will raise errors on strings given to the t() function from react hook useTranslation, which is confusing:

![](https://i.imgur.com/J19dyUl.png)

Current fix is to add t to the ignoreCallee list

```yml
rules:
   i18next/no-literal-string: ["error", { "ignoreCallee": ["cn", "t"] }]
```

**Comment by AdrienLemaire** [comment]
I'm also having eslint errors on react-router path values:
```ts
export const routes = [
  { path: "/discover/", element: Loadable(Discover), preload: Discover },
  { path: "/login/", element: Loadable(Login), preload: Login },
];
```

I can dismiss it with `"ignoreProperty": ["path"]`, but it made me wondering... Is it a good idea to translate url paths? Would love to hear your take on this :)

**Comment by edvardchen** [comment]
> By default, i18next/no-literal-string will raise errors on strings given to the t() function from react hook useTranslation, which is confusing:
> 
> 
> 
> ![](https://i.imgur.com/J19dyUl.png)
> 
> 
> 
> 
> 
> Current fix is to add t to the ignoreCallee list
> 
> 
> 
> ```yml
> 
> rules:
> 
>    i18next/no-literal-string: ["error", { "ignoreCallee": ["cn", "t"] }]
> 
> ```

it's reasonable to ignore `t` callee by default. And I think we should not check any strings in `className` property. I will handle them.

**Comment by edvardchen** [comment]
> I'm also having eslint errors on react-router path values:
> 
> ```ts
> 
> export const routes = [
> 
>   { path: "/discover/", element: Loadable(Discover), preload: Discover },
> 
>   { path: "/login/", element: Loadable(Login), preload: Login },
> 
> ];
> 
> ```
> 
> 
> 
> I can dismiss it with `"ignoreProperty": ["path"]`, but it made me wondering... Is it a good idea to translate url paths? Would love to hear your take on this :)

Sure, we should ignore urls. But how to detect if a string is url or not. Includes `/`?

**Comment by edvardchen** [comment]
> > By default, i18next/no-literal-string will raise errors on strings given to the t() function from react hook useTranslation, which is confusing:
> > ![](https://camo.githubusercontent.com/3a0fff25f9a6e04e43eaf49d6c1f0fc0b0f77015/68747470733a2f2f692e696d6775722e636f6d2f4a31396479556c2e706e67)
> > Current fix is to add t to the ignoreCallee list
> > ```yaml
> > rules:
> > 
> >    i18next/no-literal-string: ["error", { "ignoreCallee": ["cn", "t"] }]
> > ```
> 
> it's reasonable to ignore `t` callee by default. And I think we should not check any strings in `className` property. I will handle them.

I check the code `<div className={cm("bcd", "foo", "boo")}></div>` and the plugin would not throw any error. Is there anything I miss?
@AdrienLemaire

**Comment by edvardchen** [comment]
try version `4.0.0`

**Comment by AdrienLemaire** [comment]
@edvardchen thanks for the follow-up. 
I had switched to using `"markupOnly": true` in my eslintrc to reduce the verbosity of errors.

> Sure, we should ignore urls. But how to detect if a string is url or not. Includes /?
Sounds like a reasonable option, but if someone really wants to put a slash in his message...
